contracts/zero-ex/contracts/src/fixins/FixinProtocolFees.sol

```diff
@@ -0,0 +1,116 @@
+/*
+
+  Copyright 2020 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-erc20/contracts/src/v06/IEtherTokenV06.sol";
+import "../external/FeeCollector.sol";
+import "../features/libs/LibTokenSpender.sol";
+
+/// @dev Helpers for collecting protocol fees.
+abstract contract FixinProtocolFees {
+    bytes32 immutable feeCollectorCodeHash;
+
+    constructor() internal {
+        feeCollectorCodeHash = keccak256(type(FeeCollector).creationCode);
```

Comment:
Is the `immutable` cheaper than a `constant` in this case bc it'll fill in the final `bytes32` value instead of the `keccak256(...)?

Reply:
Yeah, my understanding is that a constant with the value `keccak256(...)` will be reevaluated at each access. (I wish Solidity just disallowed it.) But the `immutable` can be computed just once.